provider/pkg/provider/provider.go

```diff
@@ -799,9 +799,13 @@ func (p *googleCloudProvider) waitForResourceOpCompletion(
 					return resp, err
 				}
 
+				targetLink, hasTargetLink := getTargetLink(resp)
+				if !hasTargetLink {
+					targetLink, hasTargetLink = getSelfLink(resp)
+				}
+
```

Comment:
This is the key change - we look at both the target link and self-link properties to retrieve the target resource state.